signalflow/computation.go

```diff
@@ -206,11 +214,41 @@ func (c *Computation) bufferDataMessages() {
 	}
 }
 
+// Buffer up expiration messages indefinitely until another goroutine reads
+// them off of c.expirationCh, which is an unbuffered channel.
+func (c *Computation) bufferExpirationMessages() {
+	buffer := make([]*messages.ExpiredTSIDMessage, 0)
+	var nextMessage *messages.ExpiredTSIDMessage
+	for {
+		if len(buffer) > 0 {
+			if nextMessage == nil {
+				nextMessage, buffer = buffer[0], buffer[1:]
+			}
+			select {
+			case <-c.ctx.Done():
+				return
```

Comment:
do we need to drain any messages when ctx is done?

Reply:
It seems unnecessary since nobody would care about expiration messages if they are shutting down the computation.  Moreover they have most likely stopped reading from the channel anyway if they are stopping the computation which would make it hard to drain anything.